Allow for changing Java initial mem via new optional environment variable

neredera opened the pull request:
<!---  Ask yourself if this modification is something the whole userbase will benefit from, if this is a specific change for corner case functionality or plugins please look at making a Docker Mod or local script  https://blog.linuxserver.io/2019/09/14/customizing-our-containers/ -->
<!---  That if the PR is addressing an existing issue include, closes #<issue number> , in the body of the PR commit message   -->

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-unifi-controller/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

<!--- We welcome all PR’s though this doesn’t guarantee it will be accepted. -->

## Description:

This PR adds an optional environment variable MEM_STARTUP (default 512MB) to set the JVM startup memory (-Xms) to a value lower than the JVM memory limit (-Xml) that can already be set via MEM_LIMIT (default 1024MB).

## Benefits of this PR and context:
This PR allows to run the Unifi controller easily in resource limited environments like a Raspberry PI or simply reduces the memory usage for small and simple Unifi deployments.

## How Has This Been Tested?
I have tested the changes in my own unifi system with 9 APs but no Unifi switches or gateways.
Memory usage of the java process reacts as expected (mongodb memory usage is as expected unaffected).
I have no problems with MEM_STARTUP=256MB and MEM_LIMIT=512MB (half the defaults). I have not tested lower values.

## Source / References:

See notice from UBNT:
https://community.ui.com/questions/NOTICE-UniFi-Controller-Memory-Usage-5-6-20/228907d2-8d30-42a0-b97c-b14ef9767edb

bruvv commented:
Any update on this? I and many other people are having issues with high ram usage... (@TheSpad )

TheSpad commented:
Has this change been tested with MEM_STARTUP unset?

bruvv commented:
It is being recommended by unifi themselves: https://help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices

TheSpad commented:
> 
> 
> It is being recommended by unifi themselves: [help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices](https://help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices)

It may well be, that's not what I asked.

bruvv commented:
> 
> 
> > It is being recommended by unifi themselves: [help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices](https://help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices)
> 
> It may well be, that's not what I asked.

Good point, sorry.
I didn't test it.

But what I am wondering, why not use the unifi method of enabling it in the system.properties files?

TheSpad commented:
> 
> 
> > > It is being recommended by unifi themselves: [help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices](https://help.ui.com/hc/en-us/articles/115005159588-UniFi-How-to-Tune-the-Application-for-High-Number-of-UniFi-Devices)
> > 
> > 
> > It may well be, that's not what I asked.
> 
> Good point, sorry.
> I didn't test it.
> 
> But what I am wondering, why not use the unifi method of enabling it in the `system.properties` file?

We don't provide a base system.properties file, it's generated by the Unifi setup process, and we don't like to mess with existing config files unless there's no other option because we can't be sure what changes users might have made to them and we don't want to risk breaking things on an existing install.

TheSpad commented:
@neredera this looks good but from what I can understand from the not very clear linked Unifi docs, the defaults for both XMS and XMS are 1024Mb, so can you update the defaults to be 1024 and then I'm happy to merge it?

neredera commented:
> @neredera this looks good but from what I can understand from the not very clear linked Unifi docs, the defaults for both XMS and XMS are 1024Mb, so can you update the defaults to be 1024 and then I'm happy to merge it?

I changed the default and rebased the PR.

neredera commented:
> Has this change been tested with MEM_STARTUP unset?

I have the code of the PR running on my system since March with MEM_STARTUP not explicit set.
